### DISCUSSION TITLE
upgrade the turtlebot 4 from raspberry pi 4 to the raspberry pi 5

### DISCUSSION THREAD

**Discussion by dtwind62** [turtlebot/turtlebot4#393] [Ideas]
I recently purchased a Raspberry pi 5 after reading about all the new features and benchmarks. I think it would improve the performance of the turtlebot4 with the faster CPU, more memory, and better power management. I'm not sure if the Tb4  can be configured to boot from an ssd drive and eliminate the need to use an sd card. I purchased the Turtlebot4 to learn about robotics, so I'm not as knowledgeable as a lot of people around here about turtlebots. I did try to install my Raspberry pi 5 and it wouldn't boot from the same sd card,  the official turtlebot4 ROS2 Humble image I use on my robot. 
My questions are: 
1. can I easily configure the official turtlebot4 ROS2 Humble image I'm using to boot from the Raspberry pi 4 to work on the Raspberry pi 5?
2. Is it feasable to completely replace the sd card on the robot with an ssd drive?
3. Would it be useful to add an RTC to the turtlebot4. I also purchased an RTC battery for my Raspberry pi 5?

**Comment by hilary-luo** [comment]
You should be able to run the turtlebot4 software on whatever ubuntu device you would like with whatever drive you would like. Please follow the instructions in the read me of the [setup repo](https://github.com/turtlebot/turtlebot4_setup) to configure it without using the image. Essentially you will need to set it up with ubuntu and then install the package and run the setup script. Currently the setup tool does not set the /etc/chrony/chrony.conf file correctly though so please manually fix that file after running the script. 

An RTC can be useful depending on your use case but I don't think it is necessary for most applications. Generally the pi will just sync time to the internet or you can set it up to sync time with another device on your network.

**Reply by slowrunner** [reply]
>Currently the setup tool does not set the /etc/chrony/chrony.conf file correctly though so please manually fix that file after running the script.

Just noticed that is a post from April 2024, perhaps in January 2025 the need to "manually fix that file" is no longer needed?

The jazzy script  
- moves /etc/turtlebot4/chrony.conf to /etc/chrony/chrony.conf  

while the humble script:  
+ tests if /etc/turtlebot4/chrony.conf exists, and if so moves it to /etc/chrony/chrony.conf  

If the file exists the effect is the same, if the file does not exist shouldn't the effect be the same (the mv fails)?  


<img width="1296" alt="Screenshot 2025-01-11 at 4 01 55 PM" src="https://github.com/user-attachments/assets/f4a3671c-bfa8-4650-9b61-f473e99c4dd9" />

**Reply by hilary-luo** [reply]
@slowrunner That default comparison view that you are looking at does not compare code in the current humble and jazzy branches. It is showing that if you tried to merge the humble branch into jazzy, which commits would it attempt to apply. So that comparison that you see is the change that was done on Humble (old humble before the commit vs new humble after that commit). This has no bearing on whether or not the merge would be relevant to or successfully applied to Jazzy. This can be further understood based on the fact that if you switch which branch is base and which branch is compare, that you will see a very different result. If you do a direct code comparison [(Two-dot Git diff comparison)](https://github.com/turtlebot/turtlebot4/discussions/393#discussioncomment-11808608) or outside of github, you will see that that file is currently identical across Humble and Jazzy.

There were slight differences in the commit on the Jazzy vs Humble branches and therefore it does not recognize that both commits had the same result across the two branches.

**Comment by CycleMark** [comment]
Did you get this to work? Just trying to get Jazzy working on a RPi 5. Whilst the OS/PI is okay the TB4 (Lite) doesn't seem to have enough power from the Create board to power it.

**Reply by slowrunner** [reply]
I think you need the following:

- To see the value of usb_max_current_enable    
  (set in /boot/firmware/config.txt or if not then by OS at boot time)  
  add to /boot/firmware/config.txt:  usb_current_limit=1A6  
```
tovli@TB5WaLI:~$ vcgencmd get_config usb_max_current_enable
usb_max_current_enable=1  (if it is 0, you need to add the line to your config.txt file)
```

**Reply by CycleMark** [reply]
I've routed additional power from the 18v line on the create board via a regulator. Seems to have settled it.

If you have any ideas on this https://github.com/turtlebot/turtlebot4/issues/515 I'd be grateful.

**Reply by shuhaowu** [reply]
How did you route the power?

**Comment by manu042k** [comment]
I am trying out same things here, I’ve set up everything, and it seems to be working fine. I’m able to receive `cmd_vel` on the Raspberry Pi 5, but the bot isn’t moving. 
Any help would be useful.

**Reply by slowrunner** [reply]
Suggest open a new thread for specifically your issue - The OP was April 2024), with more specifics like 

```
printenv | grep ROS

ps -ef | grep ros

ros2 node list
```
